React/Modules/RCTUIManager.m

```diff
@@ -1053,6 +1059,16 @@ - (void)_manageChildren:(NSNumber *)containerTag
   }];
 }
 
+- (void)synchronouslyUpdateViewOnUIThread:(NSNumber *)reactTag
+                                 viewName:(NSString *)viewName
+                                    props:(NSDictionary *)props
+{
+  RCTAssertMainQueue();
+  RCTComponentData *componentData = _componentDataByName[viewName];
+  UIView *view = _viewRegistry[reactTag];
+  [componentData setProps:props forView:view];
```

Comment:
It seems like you've added that restriction at the Animated ViewManager logic, but this code is not aware of it. I'm worried that 3rd party view manager authors may use this API incorrectly.

Perhaps we could have this use a separate API and in RCTComponentData as well and verify that the props you're setting are main-thread only?